### PR TITLE
Metricfrequencyprocessor config and code cleanup

### PR DESCRIPTION
**Pull request by ppawelecsumo**

### CODE REVIEW

File: pkg/processor/metricfrequencyprocessor/config.go [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import "go.opentelemetry.io/collector/config"
+
+// Config defines configuration for Source processor.
+type Config struct {
+	*config.ProcessorSettings `mapstructure:"-"`
+
+	MinPointAccumulationSeconds           int     `mapstructure:"min_point_accumulation_seconds"`
+	ConstantMetricsReportFrequencySeconds int64   `mapstructure:"constant_metrics_report_frequency_seconds"`
+	LowInfoMetricsReportFrequencySeconds  int     `mapstructure:"low_info_metrics_report_frequency_seconds"`
+	MaxReportFrequencySeconds             int     `mapstructure:"max_report_frequency_seconds"`
```
Comment (sumo-drosiek) [inline pkg/processor/metricfrequencyprocessor/config.go:12]: `time.Duration` can be used for time types

File: pkg/processor/metricfrequencyprocessor/metriccache.go [hunk 2]
```diff
@@ -45,13 +43,13 @@ func (mc *metricCache) Register(name string, dataPoint pdata.NumberDataPoint) {
...
 	if !exists {
-		newCache := newCache()
+		newCache := mc.newCache()
 		mc.internalCaches[name] = newCache
 		internalCache = newCache
 	}
 
 	key := dataPoint.Timestamp().String()
-	value := &DataPoint{Timestamp: dataPoint.Timestamp(), Value: dataPoint.Value()}
+	value := &DataPoint{Timestamp: dataPoint.Timestamp(), Value: dataPoint.DoubleVal()}
```
Comment (sumo-drosiek) [inline pkg/processor/metricfrequencyprocessor/metriccache.go:52]: we should support `IntVal` as well

File: pkg/processor/metricfrequencyprocessor/config.go (hunk 1 above)
Comment (sumo-drosiek) [reply pkg/processor/metricfrequencyprocessor/config.go:12]: https://github.com/SumoLogic/sumologic-otel-collector/blob/main/pkg/extension/sumologicextension/config.go#L52
https://github.com/SumoLogic/sumologic-otel-collector/blob/main/pkg/extension/sumologicextension/extension.go#L71

File: pkg/processor/metricfrequencyprocessor/metriccache.go (hunk 2 above)
Comment (ppawelecsumo) [reply pkg/processor/metricfrequencyprocessor/metriccache.go]: Support added with `getVal` function and NaN filtering.

File: pkg/processor/metricfrequencyprocessor/config.go (hunk 1 above)
Comment (ppawelecsumo) [reply pkg/processor/metricfrequencyprocessor/config.go]: Handy, I forgot about it. I've switched to `time.Duration`. I'll update readme after we merge this PR.

Comment (pmalek-sumo) [reply pkg/processor/metricfrequencyprocessor/config.go]: Can we update the README in this PR so that the change is "atomic"?

**Comment by ppawelecsumo** [comment]
Something broke with compare here. Reopening.

**Pull request closed without merging**